mbed_flasher/flash.py

```diff
@@ -140,101 +154,57 @@ def flash_multiple(self, build, platform_name,
-                        return EXIT_CODE_NO_PLATFORM_GIVEN
+            return_code = self._verify_platform_coherence(device_mapping_table)
+            if return_code:
+                return return_code
 
         if isinstance(target_ids_or_prefix, list):
-            for tid in target_ids_or_prefix:
-                for item in device_mapping_table:
-                    if platform_name:
-                        if item['platform_name'] != platform_name:
-                            # skipping boards that do not match with specified platform
-                            continue
-                    if item['target_id'] == tid:
-                        aux_device_mapping_table.append(item)
+            aux_device_mapping_table = Flash._map_by_target_id(
+                device_mapping_table=device_mapping_table,
+                platform_name=platform_name,
+                target_ids=target_ids_or_prefix)
+        elif target_ids_or_prefix:
+            aux_device_mapping_table = Flash._map_by_prefix(
+                device_mapping_table=device_mapping_table,
+                platform_name=platform_name,
+                prefix=target_ids_or_prefix)
         else:
-            if target_ids_or_prefix:
-                if len(target_ids_or_prefix) >= 1:
-                    for item in device_mapping_table:
-                        if platform_name:
-                            if item['platform_name'] != platform_name:
-                                # skipping boards that do not match
-                                # with specified platform
-                                continue
-                        if item['target_id'].startswith(str(target_ids_or_prefix)):
-                            aux_device_mapping_table.append(item)
-            else:
-                for item in device_mapping_table:
-                    if platform_name:
-                        if item['platform_name'] == platform_name:
-                            aux_device_mapping_table.append(item)
+            aux_device_mapping_table = Flash._map_by_platform(
+                device_mapping_table=device_mapping_table,
+                platform_name=platform_name)
 
-        if len(aux_device_mapping_table) > 0:
+        if aux_device_mapping_table:
             device_mapping_table = aux_device_mapping_table
 
-        device_count = len(device_mapping_table)
-        if device_count == 0:
+        if not device_mapping_table:
             self.logger.error('no devices to flash')
             return EXIT_CODE_COULD_NOT_MAP_TARGET_ID_TO_DEVICE
+
         self.logger.debug(device_mapping_table)
 
-        # python 3 compatibility
-        # pylint: disable=superfluous-parens
-        print('Going to flash following devices:')
+        self.logger.info('Going to flash following devices:')
         for item in device_mapping_table:
-            print(item['target_id'])
-        retcodes = 0
-        if method == 'pyocd' and platform.system() != 'Windows':
```

Comment:
That was obsolete code.

If you look at the original code you'll notice that both branches of that if-else had basically same functionality.